From c7c78aba7e4688754bf21c641e2b85a8cc081a6f Mon Sep 17 00:00:00 2001
From: chenyutao <chenyutao2@huawei.com>
Date: Tue, 2 Jun 2026 21:48:41 +0800
Subject: [PATCH] urma: add dfx log for bondp implement.

Signed-off-by: chenyutao <chenyutao2@huawei.com>
(cherry picked from commit 8ee5db650c2e56de8259a40ea8a8cdb7a9c0bb1c)
---
 src/urma/lib/urma/bond/bondp_hash_table.c   |  2 ++
 src/urma/lib/urma/bond/bondp_health_check.c | 19 +++++++++++++------
 src/urma/lib/urma/bond/bondp_netlink.c      |  8 +++++++-
 src/urma/lib/urma/bond/bondp_provider_ops.c | 10 +++++++---
 4 files changed, 29 insertions(+), 10 deletions(-)

diff --git a/src/urma/lib/urma/bond/bondp_hash_table.c b/src/urma/lib/urma/bond/bondp_hash_table.c
index 7ddfdba5..fb05cd73 100644
--- a/src/urma/lib/urma/bond/bondp_hash_table.c
+++ b/src/urma/lib/urma/bond/bondp_hash_table.c
@@ -11,6 +11,8 @@ int bondp_hash_table_create(bondp_hash_table_t *tbl, uint32_t size,
 {
     (void)pthread_rwlock_init(&tbl->lock, NULL);
     if (ub_hmap_init(&tbl->hmap, size) != 0) {
+        URMA_LOG_ERR("Failed to init hash map, size=%u, errno=%d\n", size, errno);
+        (void)pthread_rwlock_destroy(&tbl->lock);
         return -1;
     }
     tbl->free_f = free_f;
diff --git a/src/urma/lib/urma/bond/bondp_health_check.c b/src/urma/lib/urma/bond/bondp_health_check.c
index 432b1d4b..d99f597c 100644
--- a/src/urma/lib/urma/bond/bondp_health_check.c
+++ b/src/urma/lib/urma/bond/bondp_health_check.c
@@ -224,7 +224,8 @@ static int bondp_register_health_ctx_global(bondp_context_t *bond_ctx)
 
     bondp_health_ctx_node_t *new_node = calloc(1, sizeof(bondp_health_ctx_node_t));
     if (new_node == NULL) {
-        URMA_LOG_ERR("Failed to alloc health ctx node\n");
+        URMA_LOG_ERR("Failed to alloc health ctx node, dev=%s, eid_idx=%u\n",
+                     bond_ctx->v_ctx.dev->name, bond_ctx->v_ctx.eid_index);
         return -1;
     }
 
@@ -1624,19 +1625,22 @@ int bondp_create_health_check_ctx(bondp_context_t *bond_ctx)
     }
 
     if (pthread_spin_init(&health->event_lock, PTHREAD_PROCESS_PRIVATE) != 0) {
-        URMA_LOG_ERR("Failed to init health event lock\n");
+        URMA_LOG_ERR("Failed to init health event lock, dev=%s, eid_idx=%u\n",
+                     bond_ctx->v_ctx.dev->name, bond_ctx->v_ctx.eid_index);
         return -1;
     }
 
     if (bondp_hash_table_create(&health->task_table, BONDP_HEALTH_TASK_TABLE_SIZE,
         bondp_health_task_comp_f, bondp_health_task_free_f, bondp_health_task_hash_f) != 0) {
-        URMA_LOG_ERR("Failed to create health task table\n");
+        URMA_LOG_ERR("Failed to create health task table, dev=%s, eid_idx=%u\n",
+                     bond_ctx->v_ctx.dev->name, bond_ctx->v_ctx.eid_index);
         goto ERR_EVENT_LOCK;
     }
 
     health->health_check_fd = eventfd(0, EFD_NONBLOCK | EFD_CLOEXEC);
     if (health->health_check_fd < 0) {
-        URMA_LOG_ERR("Failed to create health_check_fd, errno=%d\n", errno);
+        URMA_LOG_ERR("Failed to create health eventfd, dev=%s, eid_idx=%u, errno=%d\n",
+                     bond_ctx->v_ctx.dev->name, bond_ctx->v_ctx.eid_index, errno);
         goto DEL_TASK_TABLE;
     }
 
@@ -1644,12 +1648,15 @@ int bondp_create_health_check_ctx(bondp_context_t *bond_ctx)
     ev.data.ptr = (void *)bond_ctx;
     if (epoll_ctl(global_ctx->health_thread_ctx.health_epoll_fd,
         EPOLL_CTL_ADD, health->health_check_fd, &ev) != 0) {
-        URMA_LOG_ERR("Failed to add ctx async fd to health epoll, errno=%d\n", errno);
+        URMA_LOG_ERR("Failed to add health fd to epoll, dev=%s, eid_idx=%u, fd=%d, epoll_fd=%d, errno=%d\n",
+                     bond_ctx->v_ctx.dev->name, bond_ctx->v_ctx.eid_index,
+                     health->health_check_fd, global_ctx->health_thread_ctx.health_epoll_fd, errno);
         goto DEL_FD;
     }
 
     if (bondp_register_health_ctx_global(bond_ctx) != 0) {
-        URMA_LOG_ERR("Failed to register health ctx globally\n");
+        URMA_LOG_ERR("Failed to register health ctx globally, dev=%s, eid_idx=%u\n",
+                     bond_ctx->v_ctx.dev->name, bond_ctx->v_ctx.eid_index);
         goto DEL_EPOLL;
     }
 
diff --git a/src/urma/lib/urma/bond/bondp_netlink.c b/src/urma/lib/urma/bond/bondp_netlink.c
index 65bf0cc5..6d8e908d 100644
--- a/src/urma/lib/urma/bond/bondp_netlink.c
+++ b/src/urma/lib/urma/bond/bondp_netlink.c
@@ -113,6 +113,8 @@ static int bondp_nl_parse_cb(struct nl_msg *nlmsg, void *arg)
 
 int bondp_nl_init(void)
 {
+    int ret = 0;
+
     pthread_mutex_lock(&g_bondp_nl_ctx.lock);
     if (g_bondp_nl_ctx.sock != NULL) {
         pthread_mutex_unlock(&g_bondp_nl_ctx.lock);
@@ -122,13 +124,15 @@ int bondp_nl_init(void)
     struct nl_sock *sock = nl_socket_alloc();
     if (sock == NULL) {
         pthread_mutex_unlock(&g_bondp_nl_ctx.lock);
+        URMA_LOG_ERR("Failed to allocate netlink socket.\n");
         return -ENOMEM;
     }
 
-    int ret = genl_connect(sock);
+    ret = genl_connect(sock);
     if (ret < 0) {
         nl_socket_free(sock);
         pthread_mutex_unlock(&g_bondp_nl_ctx.lock);
+        URMA_LOG_ERR("Failed to connect generic netlink, ret=%d\n", ret);
         return ret;
     }
 
@@ -137,6 +141,7 @@ int bondp_nl_init(void)
         nl_close(sock);
         nl_socket_free(sock);
         pthread_mutex_unlock(&g_bondp_nl_ctx.lock);
+        URMA_LOG_ERR("Failed to resolve netlink family '%s', ret=%d\n", UBCORE_GENL_FAMILY_NAME, ret);
         return ret;
     }
 
@@ -144,6 +149,7 @@ int bondp_nl_init(void)
     g_bondp_nl_ctx.sock = sock;
     g_bondp_nl_ctx.genl_id = ret;
     pthread_mutex_unlock(&g_bondp_nl_ctx.lock);
+    URMA_LOG_INFO("Bond netlink initialized, genl_id=%d\n", ret);
     return 0;
 }
 
diff --git a/src/urma/lib/urma/bond/bondp_provider_ops.c b/src/urma/lib/urma/bond/bondp_provider_ops.c
index 0a3166c4..5079bc99 100644
--- a/src/urma/lib/urma/bond/bondp_provider_ops.c
+++ b/src/urma/lib/urma/bond/bondp_provider_ops.c
@@ -294,6 +294,7 @@ urma_status_t bondp_init(urma_init_attr_t *conf)
         g_bondp_global_ctx = NULL;
         return URMA_FAIL;
     }
+    URMA_LOG_INFO("Bond provider initialized successfully.\n");
     return URMA_SUCCESS;
 }
 
@@ -434,12 +435,12 @@ static int set_fd_noblock(int fd)
     int ret, flags;
     flags = fcntl(fd, F_GETFL);
     if (flags == -1) {
-        URMA_LOG_ERR("flags=%d\n", flags);
+        URMA_LOG_ERR("Failed to get fd flags, fd=%d, errno=%d\n", fd, errno);
         return -1;
     }
     ret = fcntl(fd, F_SETFL, flags | O_NONBLOCK);
     if (ret != 0) {
-        URMA_LOG_ERR("ret=%d\n", ret);
+        URMA_LOG_ERR("Failed to set fd nonblocking, fd=%d, errno=%d\n", fd, errno);
         return ret;
     }
     return 0;
@@ -536,6 +537,8 @@ static int bondp_create_pcontext(bondp_context_t *bdp_ctx, bondp_bonding_mode_t
 
         int fd = ctx->async_fd;
         if (set_fd_noblock(fd) != 0) {
+            URMA_LOG_ERR("Failed to set async fd nonblocking, dev=%s, eid_idx=%u, fd=%d\n",
+                         m->dev->name, m->eid_index, fd);
             return -1;
         }
         struct epoll_event ev = {
@@ -544,7 +547,8 @@ static int bondp_create_pcontext(bondp_context_t *bdp_ctx, bondp_bonding_mode_t
             .data.ptr = (void *)ctx,
         };
         if (epoll_ctl(bdp_ctx->v_ctx.async_fd, EPOLL_CTL_ADD, fd, &ev) != 0) {
-            URMA_LOG_ERR("failed to add fd=%u, errno=%d.\n", fd, errno);
+            URMA_LOG_ERR("Failed to add fd to epoll, dev=%s, eid_idx=%u, fd=%d, epoll_fd=%d, errno=%d\n",
+                         m->dev->name, m->eid_index, fd, bdp_ctx->v_ctx.async_fd, errno);
             return -1;
         }
     }
-- 
2.43.0

